### PR TITLE
Add support for Bearer token auth registries

### PR DESCRIPTION
**Pull request by lcarva**
Still needs unit tests, but I'd really appreciate some early feedback.

I've tested this against multiple container registries: Pulp, OpenShift Registry, Quay.io, and Proxy to Quay.io.

### CODE REVIEW

File: atomic_reactor/auth.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            return self.handle_401(response, repo, **kwargs)
+
+        response.register_hook('response', handle_401_with_repo)
+        return response
+
+    def handle_401(self, response, repo, **kwargs):
+        """Fetch Bearer token and retry."""
+
+        if response.status_code != requests.codes.unauthorized:
+            return
```
Comment (rcerven) [inline atomic_reactor/auth.py]: this should be : "return response"

Comment (lcarva) [reply atomic_reactor/auth.py]: Good catch! Fixed it.

**Comment by lcarva** [comment]
Not exactly sure why unit tests seems to pass only for python 2 on fedora 28. Will debug further in the morning.

**Comment by lcarva** [comment]
Issue was with, old, version of responses being used. Updated `tests/requirements.txt` to use a newer one.
All tests are passing now.

**Comment by lcarva** [comment]
> Please update the docs; the rest looks good.

Ooops! Forgot about it. All set now.

**Comment by lcarva** [comment]
Updated release notes draft:
> - Support for registries that use Bearer token authentication

**Pull request merged**